sycl/source/detail/program_impl.cpp

```diff
@@ -197,6 +194,13 @@ program_impl::program_impl(ContextImplPtr Context,
     MState = program_state::linked;
     MLinkOptions = "";
     MBuildOptions = Options;
+    break;
+  default:
+    // TODO: assert() will be reverted when UR_PROGRAM_BINARY_TYPE_EXECUTABLE
```

Comment:
@aelovikov-intel proposed another place for the assert ([link](https://github.com/intel/llvm/pull/10452/files#r1267379908)), can you implement his option?